industrial_taxonomy/pipeline/glass_clusters/flow.py

```diff
@@ -0,0 +1,129 @@
+"""Flow to cluster glass companies into text clusters
+"""
+
+from typing import Dict, List, Tuple
+from metaflow import FlowSpec, JSONType, project, step, Parameter
+from industrial_taxonomy.pipeline.glass_clusters.hSBM_Topicmodel.sbmtm import sbmtm
+
+
+@project(name="industrial_taxonomy")
+class ClusterGlass(FlowSpec):
+    """Create sectoral corpora of glass descriptions
+    and cluster them into text sectors
+
+    Attributes:
+        min_sector_size: minimum sector size
+        assigned_shares = share of companies to assign to cluster
+        sectors: sectors we train the model on
+        sectors_corpora: tokenised descriptions by sector
+        models: topsbm models we will use for robustness / sanity checks etc
+        clusters: lookup between company ids and clusters they are assigned to
+            clusters are named as {SIC_4 the company belongs to}_{topsbm cluster id}
+    """
+
+    min_sector_size: int
+    assigned_shares: List[float]
+    sectors_corpora: List[Dict[str, Dict[str, List[str]]]]
+    sectors: List[str]
+    clusters: List[Tuple[int, str]]
+    models: Dict[str, Dict[str, sbmtm]]
+
+    min_sector_size = Parameter(
+        "min-sector-size", help="minimum sector size", default=1000
+    )
+
+    test_mode = Parameter("test-mode", help="Run in test mode", default=True)
+
+    assigned_shares = Parameter(
+        "assigned-shares",
+        help="share of companies to assign",
+        type=JSONType,
+        default='[10,100,500,0.5,"all"]',
+    )
+
+    @step
+    def start(self):
+        """Start flow"""
+
+        self.next(self.make_sector_corpora)
+
+    @step
+    def make_sector_corpora(self):
+        """Make the sector corpora"""
+        from industrial_taxonomy.pipeline.glass_clusters.utils import (
+            make_sector_corpora,
+        )
+
+        sector_corpora = make_sector_corpora(min_sector_size=self.min_sector_size)
+
+        all_sectors_corpora = [
+            [sector, corp]
+            for sector, corp in zip(
+                list(sector_corpora.keys()), sector_corpora.values()
+            )
+        ]
```

Comment:
I think all this is just...

```suggestion
        all_sectors_corpora = list(sector_corpora.items())
```
?

Reply:
Is there any reason why it can't remain as a `dict`?

If the answer is for the test_mode truncation then you can do

```python
from toolz import take

self.sectors_corpora = dict(take(3, sectors_corpora.items())
```